lightning/src/ln/channel.rs

```diff
@@ -4354,28 +4355,43 @@ mod tests {
 		PublicKey::from_secret_key(&secp_ctx, &SecretKey::from_slice(&hex::decode(hex).unwrap()[..]).unwrap())
 	}
 
-	#[test]
-	fn channel_reestablish_no_updates() {
-		let feeest = TestFeeEstimator{fee_est: 15000};
-		let logger = test_utils::TestLogger::new();
-		let secp_ctx = Secp256k1::new();
-		let mut seed = [0; 32];
-		let mut rng = thread_rng();
-		rng.fill_bytes(&mut seed);
-		let network = Network::Testnet;
-		let keys_provider = test_utils::TestKeysInterface::new(&seed, network);
+	struct ChanCfg {
+		fee_est: TestFeeEstimator,
+		logger: test_utils::TestLogger,
+		keys_mgr: test_utils::TestKeysInterface,
```

Comment:
Do these need to be different for each side of the channel? If not, I'm wondering if we can dispense with `ChanCfg` and `create_chan_cfgs` entirely and simply pass these directly into `create_channels`.

Reply:
Hm, I see what you mean. Well, one way they _should_ be different is having separate loggers so we can distinguish their logs (I've added IDs to each logger now).

Open to suggestions here -- one option would be to do what you suggest, but pass in two loggers instead of one.

Going this route could potentially limit flexibility with testing stuff like e.g. the channel sides  disagreeing on fees, or other cases like that. But that's not a problem right now, and I've heard the quote "premature optimization is the root of all evil" so open to whatever :P

Reply:
Gotcha! I'd argue then that `create_chan_cfgs` should take an id for logging (pubkey instead?), return a single config, and be called twice where needed. Then in the future additional arguments could be added for the fee estimation, for instance.

This would also simplify the implementation by removing the loop, match, and mutable `Option` variables.